modelzoo/admin.py

```diff
@@ -0,0 +1,475 @@
+        try:
+            self.auth = UserAuth(
+                self, username=username, password=password, token=token
+            )
+        except AuthenticationException as e:
+            self.auth = None
+
+    def connect(self, address: typing.Optional[str] = "grpc.modelzoo.live"):
+        self.address = self.address if address is None else address
+        channel = grpc.insecure_channel(self.address)
+        self.conn = ModelStub(channel)
+
+    def authenticate(
+        self,
+        username: typing.Optional[str] = "",
+        password: typing.Optional[str] = "",
+        token: typing.Optional[str] = "",
+    ) -> None:
+        try:
+            self.auth = UserAuth(
+                self, username=username, password=password, token=token
+            )
+        except InsufficientCredentialsException:
+            raise ModelZooConnectionException(
+                "At least one of [(Username, Password), Token] are required."
+            )
+        except InvalidTokenException:
+            raise ModelZooConnectionException("The token provided is not valid.")
+
+    def _authenticate(self, username: str, password: str) -> str:
+        if self.conn is None:
+            raise self.conn_error
+        resp = self.conn.Authenticate(
+            AuthenticationRequest(username=username, password=password)
+        )
+        if resp.success:
+            return resp.token
+        else:
+            raise ModelZooConnectionException(resp.msg)
+
+    def get_token(self):
+        if self.auth is None:
+            raise ModelZooConnectionException(
+                "Please authenticate before calling this method."
+            )
+        if self.conn is None:
+            raise self.conn_error
+        return self.auth.token
+
+    def get_admin_token(self):
+        if self.auth is None:
+            raise ModelZooConnectionException(
+                "Please authenticate before calling this method."
+            )
+        if self.conn is None:
+            raise self.conn_error
+        if self.auth.admin_token is None:
+            raise ModelZooConnectionException(
+                "Please authenticate with username and password to recieve your administrative token."
+            )
+        return self.auth.admin
+
+    def register_user(self, username: str, password: str, email: str) -> dict:
+        if self.conn is None:
+            raise self.conn_error
+        resp = self.conn.RegisterUser(
+            RegisterUserRequest(username=username, password=password, email=email)
+        )
+        if resp.success:
+            return {
+                "Username": username,
+                "Password": password,
+                "Admin Token": resp.token,
+            }
+        else:
+            raise ModelZooConnectionException(resp.msg)
+
+    def register_token(
+        self,
+        username: typing.Optional[str] = "",
+        password: typing.Optional[str] = "",
+        token: typing.Optional[str] = "",
```

Comment:
this token is old token? we should enforce new token can only be registered with user_name + password because we don't want tokens cascading creation.